generator/internal/scaffold_generator.cc

```diff
@@ -138,6 +147,62 @@ std::map<std::string, std::string> ScaffoldVars(
                      "to change without notice.\n\nPlease,"
                    : "While this library is **GA**, please";
 
+  // Try to load the service config YAML file. On failure just return the
+  // existing vars.
+  auto const config_file = vars.find("configFile");
+  auto const directory = vars.find("directory");
+  if (config_file == vars.end() || directory == vars.end()) {
+    GCP_LOG(WARNING) << "Missing directory and/or YAML config file name for: "
+                     << service.service_proto_path();
+    return vars;
+  }
+
+  auto const path =
+      googleapis_path + "/" + directory->second + "/" + config_file->second;
+  auto status = google::cloud::internal::status(path);
+  if (!exists(status)) {
+    GCP_LOG(WARNING) << "Cannot find YAML service config file (" << path
+                     << ") for: " << service.service_proto_path();
+    return vars;
+  }
+  auto config = YAML::LoadFile(path);
+  if (config.Type() != YAML::NodeType::Map) {
+    GCP_LOG(WARNING) << "Error loading YAML config file (" << path
+                     << ") for: " << service.service_proto_path()
+                     << "  error=" << config.Type();
+    return vars;
+  }
+  auto publishing = config["publishing"];
+  // This error is too common at the moment. Most libraries lack a
+  // 'publishing' section.
+  if (publishing.Type() != YAML::NodeType::Map) return vars;
+
+  for (std::string const name : {
+           "api_short_name",
+           "documentation_uri",
+           "new_issue_uri",
+       }) {
+    auto node = publishing[name];
+    if (node.Type() != YAML::NodeType::Scalar) continue;
+    auto value = node.as<std::string>();
+    if (value.empty()) continue;
+    vars[name] = std::move(value);
+  }
+  // The YAML configuration includes a link to create new issues. If possible,
+  // convert that to a link to list issues, which is what we want to generate.
+  auto l = vars.find("new_issue_uri");
+  if (l != vars.end()) {
+    auto issue_tracker = l->second;
+    auto const re = std::regex(
+        R"re((https://issuetracker.google.com/issues).*[^a-z]component=([0-9]*).*)re");
+    std::smatch match;
+    if (std::regex_match(issue_tracker, match, re)) {
+      issue_tracker = absl::StrCat(
+          match[1].str(), "?q=componentid:", match[2].str(), " status=open");
```

Comment:
`' '` in the link seems weird, but :shrug:

Reply:
Fixed